modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapperTestCase.java

```diff
@@ -47,6 +51,22 @@ public void recursiveParameterType() {
     }
   }
 
+  @Test
+  @Description("Ensure JDK internal fields are filtered out as we can't use reflection on them in Java 17 and above")
+  public void filterInternalJdkFields() {
+    TypeWrapper type = new TypeWrapper(LocalDateTime.class, new DefaultExtensionsTypeLoaderFactory()
+        .createTypeLoader(Thread.currentThread().getContextClassLoader()));
+    List<FieldElement> fields = type.getFields();
+    assertThat(fields.stream().noneMatch(f -> f.getField().get().getName().equals("serialVersionUID")),
+               is(true));
+    assertThat(fields.isEmpty(), is(true));
+    // verify that we are not filtering out fields that are Java types in a custom class
+    type = new TypeWrapper(PersonalInfo.class, new DefaultExtensionsTypeLoaderFactory()
+        .createTypeLoader(Thread.currentThread().getContextClassLoader()));
+    assertThat(type.getFields().stream().anyMatch(f -> f.getField().get().getName().equals("dateOfConception")),
```

Comment:
Indirect assertion. The correct assertion would be checking that the list is not empty but none of the fields come from a java package